src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -289,56 +291,79 @@ public void insertBufferedRows() {
     }
   }
 
-  // should we rely on precommits frequency to give us the last committed offset token?
-  // or should we have a BG thread polling on every channel to update the Atomic long and simply
-  // return the atomic long's current value
-  // For now we will rely on preCommit API to fetch the inserted offsets into Snowflake
   // TODO: SNOW-529755 POLL committed offsets in backgraound thread
   /**
    * Get committed offset from Snowflake. It does an HTTP call internally to find out what was the
    * last offset inserted.
    *
    * <p>If committedOffset fetched from Snowflake is null, we would return -1(default value of
    * committedOffset) back to original call. (-1) would return an empty Map of partition and offset
-   * back to kafka. (We)
+   * back to kafka.
    *
-   * <p>Check {@link com.snowflake.kafka.connector.SnowflakeSinkTask#preCommit(Map)}
+   * <p>Else, we will convert this offset and return the offset which is safe to commit inside Kafka
+   * (+1 of this returned value).
    *
-   * <p>Else, we will convert this offset and return the offset which is safe to commit inside
-   * Kafka.
+   * <p>Check {@link com.snowflake.kafka.connector.SnowflakeSinkTask#preCommit(Map)}
    *
-   * @return offsetToken present in Snowflake, else -1
+   * @return (offsetToken present in Snowflake + 1), else -1
    */
-  public long getCommittedOffset() {
-    LOGGER.debug(
-        "Fetching last committed offset for partition channel:{}",
-        this.channel.getFullyQualifiedName());
-    String lastOffsetCommitted = channel.getLatestCommittedOffsetToken();
-    LOGGER.info(
-        "Last committed offset for partition channel:{} is :{}",
-        this.channel.getFullyQualifiedName(),
-        lastOffsetCommitted);
-    if (Strings.isNullOrEmpty(lastOffsetCommitted)) {
-      return committedOffset.get();
+  public long getOffsetSafeToCommitToKafka() {
+    final long committedOffsetInSnowflake = fetchLatestCommittedOffsetFromSnowflake();
+    if (committedOffsetInSnowflake == NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE) {
+      return offsetSafeToCommitToKafka.get();
```

Comment:
could we just return -1? It's easier to understand

Reply:
Ahh yes!

Reply:
No, we cant return -1; (As above, it it init to 0)

Reply:
ok, then could we return 0 instead? (or at least add a comment saying this is always 0, otherwise the logic is confusing by only looking at the code)

Reply:
The only reason I didnt want to return 0 was it offsetSafeToCommitToKafka could be non zero. 

i.e it is possible  `fetchLatestCommittedOffsetFromSnowflake` gave us invalid token, in that case we would like to return the offset which was last sent to kafka. (which can be any number)